src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -64,14 +75,16 @@ public override bool Execute()
                 return false;
             }
 
-            List<ITaskItem> packageInformation = new List<ITaskItem>();
+            Dictionary<string, PackageItem> packageInformation = new Dictionary<string, PackageItem>();
             _packageNameRegex = new Regex(PackageNameRegex);
 
+            VersionComparer comparer = new VersionComparer(VersionComparison.VersionRelease);
```

Comment:
You only ever set this to `VersionComparer(VersionComparison.VersionRelease)`.  Can you make it an instance field and avoid passing it around everywhere?